src/utils/hash.ts

```diff
@@ -205,13 +206,14 @@ export function calculateContractAddressFromHash(
 
   const CONTRACT_ADDRESS_PREFIX = felt('0x535441524b4e45545f434f4e54524143545f41444452455353'); // Equivalent to 'STARKNET_CONTRACT_ADDRESS'
 
-  return computeHashOnElements([
+  const hash = computeHashOnElements([
     CONTRACT_ADDRESS_PREFIX,
     deployerAddress,
     salt,
     classHash,
     constructorCalldataHash,
   ]);
+  return toHex(BigInt(hash) % ADDR_BOUND);
```

Comment:
I am not sure about this return if it will behave the same as Network, but I guess it will also return it to bounded value?

Reply:
I was surprised when the issue was reported since it never popped up, so I checked with [cairo-lang](https://github.com/starkware-libs/cairo-lang/blob/v0.12.3/src/starkware/starknet/common/storage.cairo#L8), [starknet.py](https://github.com/software-mansion/starknet.py/blob/v0.18.3/starknet_py/hash/address.py#L41), and [starknet-rs](https://github.com/xJonathanLEI/starknet-rs/blob/starknet-core/v0.7.2/starknet-core/src/utils.rs#L190). They use the same approach, it just seems the out of bounds values are rare so nobody noticed.